### PR TITLE
Fix issues with custom reagent holder

### PR DESCRIPTION
**Pull request by jacklecroy**
## Changelog
:cl: That REALLY Good Soda Flavor
fix: A bun will no longer disappear if you add an ingredient to it while it is in your hand.
fix: Custom burgers will no longer disappear from adding too many ingredients.
/:cl:

### CODE REVIEW

File: code/datums/components/customizable_reagent_holder.dm [hunk 1]
```diff
@@ -28,7 +28,7 @@
 		atom/replacement,
 		fill_type,
 		ingredient_type = CUSTOM_INGREDIENT_TYPE_EDIBLE,
-		max_ingredients = INFINITY,
+		max_ingredients = 98,
```
Comment (Mothblocks) [inline code/datums/components/customizable_reagent_holder.dm:31]: Can you explain what this does to fix the bugs you've mentioned?

File: code/datums/components/customizable_reagent_holder.dm [hunk 2]
```diff
@@ -122,7 +122,7 @@
 	if(!attacker.transferItemToLoc(ingredient, atom_parent))
 		return
 	if (replacement)
-		var/atom/replacement_parent = new replacement(atom_parent.loc)
+		var/atom/replacement_parent = new replacement(get_turf(atom_parent))
```
Comment (Cyberboss) [inline code/datums/components/customizable_reagent_holder.dm:125]:
```suggestion
		var/atom/replacement_parent = new replacement(atom_parent.drop_location())
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/customizable_reagent_holder.dm (hunk 1 above)
Comment (jacklecroy) [reply code/datums/components/customizable_reagent_holder.dm:31]: overlay limit is 100

Comment (Mothblocks) [reply code/datums/components/customizable_reagent_holder.dm:31]: Replace this with `MAX_ATOM_OVERLAYS - 2` then.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Replace magic number with MAX_ATOM_OVERLAYS - 2.

**Pull request merged**